Patch by Robert Scheck <robert@fedoraproject.org> for ssldump >= 0.9b3, which is
changing the decoder table ends from 0 to -1 to match the expected value of table
search routines. Without this patch, ssldump segfaults at latest after some time
of usage when decoding unknown enumerated values. For further information, please
have a look to Red Hat Bugzilla, bug ID #747398.

--- ssldump-0.9b3/ssl/ssl.enums.c		2011-10-24 22:33:03.000000000 +0200
+++ ssldump-0.9b3/ssl/ssl.enums.c.table-stops	2011-10-24 22:34:20.000000000 +0200
@@ -500,7 +500,7 @@
 		"Finished",
 		decode_HandshakeType_Finished
 	},
-{0}
+{-1}
 };
 
 decoder cipher_suite_decoder[]={
@@ -778,7 +778,7 @@
 		"fatal",
 		decode_AlertLevel_fatal
 	},
-{0}
+{-1}
 };
 
 static int decode_AlertDescription_close_notify(ssl,dir,seg,data)
@@ -1081,7 +1081,7 @@
 		"no_renegotiation",
 		decode_AlertDescription_no_renegotiation
 	},
-{0}
+{-1}
 };
 
 decoder compression_method_decoder[]={
@@ -1145,6 +1145,6 @@
 		"dss_fixed_dh",
 		decode_client_certificate_type_dss_fixed_dh
 	},
-{0}
+{-1}
 };
 
--- ssldump-0.9b3/ssl/ssl_enum.c		2011-10-24 22:33:03.000000000 +0200
+++ ssldump-0.9b3/ssl/ssl_enum.c.table-stops	2011-10-24 22:34:44.000000000 +0200
@@ -260,7 +260,7 @@
 		"finished",
 		decode_HandshakeType_finished
 	},
-{0}
+{-1}
 };
 
 decoder cipher_suite_decoder[]={
